### ISSUE TITLE
WSL 2 GPU support from Docker Desktop Windows

### ISSUE THREAD

**Issue by Nishanksingla** [docker/for-win#11246]
Hello

I am trying to run gpu container using docker desktop on windows 10. I have followed all the steps from this URL 
https://docs.docker.com/docker-for-windows/wsl/#gpu-support

I have met all the following requirements
- Windows 10 Pro from dev channel build version 21382.1
- Nvidia GPU Titan X pascal
- NVIDIA CUDA WSL drivers from this link https://developer.nvidia.com/cuda/wsl (driver version 470)
- WSL 2 enabled in windows
-  WSL 2 backend is enabled in Docker Desktop
- Latest linux kernel
- Docker Desktop for windows version 3.3.3

I am running following docker command from **windows command prompt**

docker run --gpus all --env NVIDIA_DISABLE_REQUIRE=1 nvcr.io/nvidia/k8s/cuda-sample:nbody 5 nbody -gpu -benchmark

and I am getting this error

docker: Error response from daemon: OCI runtime create failed: container_linux.go:367: starting container process caused: process_linux.go:495: container init caused: Running hook #0:: error running hook: exit status 1, stdout: , stderr: nvidia-container-cli: initialization error: driver error: failed to process request: unknown.

So I am not understanding what's wrong.

**Comment by cpbotha** [comment]
I am seeing exactly the same behavior on my setup.

- Windows Insiders dev build 21382.1000
- NVIDIA driver 470.14
-  everything else up to date as above
- Docker for Desktop 3.3.3

I can build and run many of the CUDA samples, for example BlackScholes, on my only WSL2 instance. Also the Docker for Desktop WSL2 integration works like a charm. Only thing that does not work is the docker vGPU support.

**Comment by sumandeepb** [comment]
@Nishanksingla @cpbotha Check if this may help: https://forums.developer.nvidia.com/t/guide-to-run-cuda-wsl-docker-with-latest-versions-21382-windows-build-470-14-nvidia/178365

**Comment by sumandeepb** [comment]
I was using the same config as you, but I solved by not using Docker for Desktop, installing nvidia-docker2 et. al directly on the Ubuntu distro.

And using the downgraded (not latest) versions:

```
$ sudo apt  install nvidia-docker2:amd64=2.5.0-1 \
           libnvidia-container-tools:amd64=1.3.3-1 \
           nvidia-container-runtime:amd64=3.4.2-1 \
           libnvidia-container1:amd64=1.3.3-1 \
           nvidia-container-toolkit:amd64=1.4.2-1
```

**Comment by Nishanksingla** [comment]
Hey sumandeepb 

Thank you for the response. 
I followed these instructions 
https://docs.nvidia.com/cuda/wsl-user-guide/index.html#running-cuda

I got the same error in the ubuntu distro as well. 

As you mentioned, I downgraded the versions and it's working now inside the ubuntu distro. 
Thank you for the help

**Comment by stephen-turner** [comment]
Thanks for the reports. I'm struggling to understand this because there don't immediately seem to be any relevant changes between Desktop 3.3.1 and 3.3.3: we didn't take a new kernel for example. But we'll have a look.

**Comment by calclavia** [comment]
Getting the same issue after the upgrade to Docker 3.3.3.

**Comment by jnoelvictorino** [comment]
Is there any way to downgrade from Docker 3.3.3 to Docker 3.3.0 without losing the images?

**Comment by aiordache** [comment]
Hi @Nishanksingla. Did a driver update to 470.76 fix your issue?

**Comment by jnoelvictorino** [comment]
Docker Desktop for Windows 3.5.1 66090 worked.

**Comment by stephen-turner** [comment]
This was fixed by Nvidia, see NVIDIA/nvidia-docker#1496. Closing this ticket.
